vic/drivers/shared_all/src/compute_derived_state_vars.c

```diff
@@ -0,0 +1,234 @@
+                                    "step length to <= %f seconds, or decrease "
+                                    "the number of soil thermal nodes.",
+                                    global_param.dt,
+                                    soil_con->dz_node[1],
+                                    energy[veg][band].Cs_node[1],
+                                    energy[veg][band].kappa_node[1], dt_thresh);
+                        }
+                    }
+
+                    /* initialize layer moistures and ice contents */
+                    for (lidx = 0; lidx < options.Nlayer; lidx++) {
+                        cell[veg][band].layer[lidx].moist =
+                            moist[veg][band][lidx];
+                        for (frost_area = 0;
+                             frost_area < options.Nfrost;
+                             frost_area++) {
+                            cell[veg][band].layer[lidx].ice[frost_area] =
+                                ice[veg][band][lidx][frost_area];
+                        }
+                    }
+                    if (options.QUICK_FLUX) {
+                        ErrorFlag =
+                            estimate_layer_ice_content_quick_flux(
+                                cell[veg][band].layer,
+                                soil_con->depth, soil_con->dp,
+                                energy[
+                                    veg][band].T[0], energy[veg][band].T[1],
+                                soil_con->avg_temp, soil_con->max_moist,
+                                soil_con->expt, soil_con->bubble,
+                                soil_con->frost_fract, soil_con->frost_slope,
+                                soil_con->FS_ACTIVE);
+                    }
+                    else {
+                        ErrorFlag = estimate_layer_ice_content(
+                            cell[veg][band].layer,
+                            soil_con->Zsum_node,
+                            energy[veg][band].T,
+                            soil_con->depth,
+                            soil_con->max_moist,
+                            soil_con->expt,
+                            soil_con->bubble,
+                            soil_con->frost_fract,
+                            soil_con->frost_slope,
+                            options.Nnode,
+                            options.Nlayer,
+                            soil_con->FS_ACTIVE);
+                    }
+
+                    /* Find freezing and thawing front depths */
+                    if (!options.QUICK_FLUX && soil_con->FS_ACTIVE) {
+                        find_0_degree_fronts(&energy[veg][band],
+                                             soil_con->Zsum_node,
+                                             energy[veg][band].T,
+                                             options.Nnode);
+                    }
+                }
+            }
+        }
+    }
+
+    return(0);
```

Comment:
we've been moving away from returning an exit status unless it will be used later.  In this case, we should exit immediately if there are any errors and we can return with `void` instead.